### DISCUSSION TITLE
SV06-Plus MCU unable to connect

### DISCUSSION THREAD

**Discussion by dalester7** [bassamanator/Sovol-SV06-firmware#120] [Q&A]
Hi! I've read through the readme many times now and am not having much luck. I am wondering if it's possibly due to the finnicky nature of flashing the mobo. Here is what is going on...

I ran through all the steps, installed the on a raspberry pi 3B. I used other images --> pi legacy 32bit lite and then installed KIAUH to install klipper, moonraker, mainsail, and obico. I made sure all the files were updated through Mainsail. I'm running a very basic setup so most options were the default (single instance, etc). 

I DL the Klipper Config per method 1. Seemed to work properly.

I updated the firmware using the file klipper-v0.12.0-10-gea2f6bc0.bin, which was in /misc. When I put the card into the printer, it powers up and the screen shows the Sovol still image. I let it sit for about a minute before I powered off and removed the card. Then I connected the Pi to the printer and started it up and started the printer back up. I got the serial address and modified the printer.cfg to this:

[mcu]
# NOTE Obtain definition by "ls /dev/serial/by-id/*" or "ls /dev/serial/by-path/*"
serial: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
restart_method: command

Then I restart and firmware_restart but I continue to get the error that mcu 'mcu': unable to connect. In the klippy.log I see that it appears to be connecting to a generic address: 

mcu 'mcu': Starting serial connect
mcu 'mcu': Unable to open serial port: [Errno 2] could not open port /dev/serial/by-id/<your-mcu-id>: [Errno 2] No such file or directory: '/dev/serial/by-id/<your-mcu-id>'
mcu 'mcu': Unable to open serial port: [Errno 2] could not open port /dev/serial/by-id/<your-mcu-id>: [Errno 2] No such file or directory: '/dev/serial/by-id/<your-mcu-id>'
...
MCU error during connect
Traceback (most recent call last):
  File "/home/pi/klipper/klippy/mcu.py", line 800, in _mcu_identify
    self._serial.connect_uart(self._serialport, self._baud, rts)
  File "/home/pi/klipper/klippy/serialhdl.py", line 182, in connect_uart
    self._error("Unable to connect")
  File "/home/pi/klipper/klippy/serialhdl.py", line 61, in _error
    raise error(self.warn_prefix + (msg % params))
serialhdl.error: mcu 'mcu': Unable to connect

I verified that the correct address is set in the printer.cfg. I flashed the printer back to official Sovol SV06 Plus firmware, boots up again. Tried to flash again to the one from the /misc folder and same issue.

I'm a NOOB with 3D printing but would like to be able to use klipper so any help would be appreciated. Not sure what to try next. Thanks!!

**Comment by bassamanator** [comment]
Your changes don't appear to be saving to `printer.cfg`, figure that bit out and you should be fine.

Also make sure your user is part of the `dialout` group:
1. `groups $USER`
2. If it isnt' then:
- `sudo usermod -a -G tty $USER`

https://explainshell.com

**Comment by dalester7** [comment]
Thanks for the help. I ended up re-formatting the card and re-downloading the .bin and re-installing the firmware. It worked. I'd already done that multiple times so not really sure what exactly fixed it. But it's working now. Thanks for creating this!!

**Reply by bassamanator** [reply]
Great! Please star the repo.
